sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityVariableSet.java

```diff
@@ -38,14 +36,20 @@
      */
     public SensitivityVariableSet(String id, List<WeightedSensitivityVariable> variables) {
         this.id = Objects.requireNonNull(id);
-        this.variables = Collections.unmodifiableList(Objects.requireNonNull(variables));
+        //Use LinkedHashMap to preserve insertion order
+        this.variables = Collections.unmodifiableMap(
+                Objects.requireNonNull(variables).stream().collect(LinkedHashMap::new, (map, item) -> map.put(item.getId(), item), Map::putAll));
     }
 
     public String getId() {
         return id;
     }
 
-    public List<WeightedSensitivityVariable> getVariables() {
+    public Collection<WeightedSensitivityVariable> getVariables() {
+        return variables.values();
+    }
+
+    public Map<String, WeightedSensitivityVariable> getVariablesById() {
```

Comment:
Wouldn't it be more convenient to add directly a `getVariable(String id)`? because I don't think the map itself is of interest.

Reply:
Exposing the map gives access to the map api which can be convenient in some context (containsKey, keySet...). I can offer a getVariable(String id) though.